lib/rs/src/protocol/mod.rs

```diff
@@ -213,25 +236,19 @@ impl ProtocolHelpers {
                 //throw x
                 Err(ThriftErr::Exception)
             }
-            (fname, MessageType::MtReply, _) => {
+            (fname, _, _) => {
```

Comment:
Yea, it should be an error if the server receives an MtReply or if the client receives an MtCall, but other than that the code is exactly the same, so I merged the two for now. It doesn't feel like a high priority issue at the moment (it's not something the user of the library has to deal with), so I'll add a `TODO` for now.